wrapper/src/main/java/software/amazon/jdbc/wrapper/ConnectionWrapper.java

```diff
@@ -122,8 +124,19 @@ protected void init(
 
     this.pluginManager.init(this.pluginService, props, pluginManagerService);
 
-    this.hostListProviderService.setHostListProvider(
-        new ConnectionStringHostListProvider(props, this.originalUrl, this.hostListProviderService));
+    if (this.hostListProviderService.getDialect() instanceof TopologyAwareDatabaseCluster) {
+      this.hostListProviderService.setHostListProvider(new AuroraHostListProvider(
+          this.targetDriverProtocol,
+          this.hostListProviderService,
+          props,
+          this.originalUrl)
+      );
+    } else {
+      this.hostListProviderService.setHostListProvider(new ConnectionStringHostListProvider(
+          props,
+          this.originalUrl,
+          this.hostListProviderService));
+    }
 
     this.pluginManager.initHostProvider(
```

Comment:
After allowing plugins to set up their own host list provider, it's a perfect time to set up a host list provider mentioned in dialect (if any).